packages/iota/src/validation/outputs/aliasOutput.ts

```diff
@@ -0,0 +1,156 @@
+/**
+ * Maximum number of unlock conditions that alias output can have.
+ */
+const MAX_ALIAS_UNLOCK_CONDITIONS_COUNT = 2;
+
+/**
+ * Validate an alias output.
+ * @param aliasOutput The object to validate.
+ * @param protocolInfo The Protocol Info.
+ * @returns The validation result.
+ */
+export function validateAliasOutput(aliasOutput: IAliasOutput, protocolInfo: INodeInfoProtocol): IValidationResult {
+    const results: IValidationResult[] = [];
+
+    if (aliasOutput.type !== ALIAS_OUTPUT_TYPE) {
+        results.push({
+            isValid: false,
+            errors: [`Type mismatch in alias output ${aliasOutput.type}`]
+        });
+    }
+
+    if (bigInt(aliasOutput.amount).compare(bigInt.zero) !== 1) {
+        results.push({
+            isValid: false,
+            errors: ["Alias output amount field must be larger than zero."]
+        });
+    }
+
+    if (bigInt(aliasOutput.amount).compare(protocolInfo.tokenSupply) === 1) {
+        results.push({
+            isValid: false,
+            errors: ["Alias output amount field must not be larger than max token supply."]
+        });
+    }
+
+    if (aliasOutput.unlockConditions.length !== MAX_ALIAS_UNLOCK_CONDITIONS_COUNT) {
+        results.push({
+            isValid: false,
+            errors: [`Unlock conditions count must be equal to ${MAX_ALIAS_UNLOCK_CONDITIONS_COUNT}.`]
+        });
+    }
+
+    if (aliasOutput.unlockConditions) {
+        if (
+            !aliasOutput.unlockConditions.every(
+                uC =>
+                    uC.type === STATE_CONTROLLER_ADDRESS_UNLOCK_CONDITION_TYPE ||
+                    uC.type === GOVERNOR_ADDRESS_UNLOCK_CONDITION_TYPE
+            )
+        ) {
+            results.push({
+                isValid: false,
+                errors: ["Alias output unlock condition type of an unlock condition must define one of the following types: State Controller Address Unlock Condition and Governor Address Unlock Condition."]
+            });
+        }
+
+        aliasOutput.unlockConditions.map(unlockCondition => {
+            if ((unlockCondition.type === STATE_CONTROLLER_ADDRESS_UNLOCK_CONDITION_TYPE ||
+                unlockCondition.type === GOVERNOR_ADDRESS_UNLOCK_CONDITION_TYPE) &&
+                (unlockCondition.address.type === ALIAS_ADDRESS_TYPE &&
+                    unlockCondition.address.aliasId === aliasOutput.aliasId)) {
+                results.push({
+                    isValid: false,
+                    errors: ["Alias output Address field of the State Controller Address Unlock Condition and Governor Address Unlock Condition must not be the same as the Alias address derived from Alias ID."]
+                });
+            }
+        });
+        results.push(validateAscendingOrder(aliasOutput.unlockConditions, "Alias output", "Unlock Condition"));
+        results.push(validateUnlockConditions(aliasOutput.unlockConditions));
+    }
+
+    if (aliasOutput.nativeTokens) {
```

Comment:
Do this before unlock conditions, let's try to preserve the order of rules from the TIP. Also check ordering how is done in nftOutput.ts or foundryOutput.ts